lib/identity_cache/parent_model_expiration.rb

```diff
@@ -4,44 +4,61 @@ module ParentModelExpiration # :nodoc:
 
     included do |base|
       base.class_attribute :parent_expiration_entries
-      base.parent_expiration_entries = Set.new
+      base.parent_expiration_entries = Hash.new{ |hash, key| hash[key] = [] }
     end
-  
-    module ClassMethods
-      private
 
-      def add_parent_expiration_entry(after_action_name)
-        parent_expiration_entries << after_action_name
-      end
-    end
-    
     def expire_parent_caches
-      self.class.parent_expiration_entries.each do |parent_expiration_entry|
-        send(parent_expiration_entry)
+      parents_to_expire = {}
+      add_parents_to_cache_expiry_set(parents_to_expire)
+      parents_to_expire.each_value do |parent|
+        parent.send(:expire_primary_index)
       end
     end
 
-    def expire_parent_cache_on_changes(parent_name, foreign_key, parent_class, only_on_foreign_key_change)
-      new_parent = send(parent_name)
+    def add_parents_to_cache_expiry_set(parents_to_expire)
```

Comment:
`parent_expiration_entries` is not a Set.. not sure if we should have the suffix `_set` on the method name